src/DGtal/dec/DiscreteExteriorCalculusFactory.h

```diff
@@ -0,0 +1,191 @@
+#else // defined(DiscreteExteriorCalculusFactory_RECURSES)
+/** Prevents recursive inclusion of headers. */
+#define DiscreteExteriorCalculusFactory_RECURSES
+
+#if !defined DiscreteExteriorCalculusFactory_h
+/** Prevents repeated inclusion of headers. */
+#define DiscreteExteriorCalculusFactory_h
+
+//////////////////////////////////////////////////////////////////////////////
+#include "DGtal/dec/DiscreteExteriorCalculus.h"
+#include "DGtal/topology/DigitalSurface.h"
+//////////////////////////////////////////////////////////////////////////////
+
+namespace DGtal
+{
+
+/////////////////////////////////////////////////////////////////////////////
+// class DiscreteExteriorCalculusFactory
+/**
+ * Description of class 'DiscreteExteriorCalculusFactory' <p>
+ * \brief Aim:
+ * This class provides static members to create DEC structures from various other DGtal structures.
+ *
+ * @tparam TLinearAlgebraBackend linear algebra backend used (i.e. EigenLinearAlgebraBackend).
+ * @tparam TInteger integer type forwarded to khalimsky space.
+ */
+
+template <typename TLinearAlgebraBackend, typename TInteger = DGtal::int32_t>
+class DiscreteExteriorCalculusFactory
+{
+    // ----------------------- Standard services ------------------------------
+public:
+
+    typedef typename TLinearAlgebraBackend::DenseVector DenseVector;
+    typedef typename TLinearAlgebraBackend::DenseMatrix DenseMatrix;
+    typedef typename TLinearAlgebraBackend::SparseMatrix SparseMatrix;
+
+    BOOST_CONCEPT_ASSERT(( concepts::CInteger<TInteger> ));
+
+    BOOST_CONCEPT_ASSERT(( concepts::CDynamicVector<DenseVector> ));
+    BOOST_CONCEPT_ASSERT(( concepts::CDynamicMatrix<DenseMatrix> ));
+    BOOST_CONCEPT_ASSERT(( concepts::CDynamicMatrix<SparseMatrix> ));
+    BOOST_CONCEPT_ASSERT(( concepts::CLinearAlgebra<DenseVector, SparseMatrix> ));
+    BOOST_CONCEPT_ASSERT(( concepts::CLinearAlgebra<DenseVector, DenseMatrix> ));
+
+    /**
+     * Create DEC structure from digital set.
+     * DEC embedded and ambient dimensions are equal to digital set point dimension.
+     * Points of the set get attached to primal n-cell <-> dual 0-cell.
+     * @tparam TDigitalSet type of digital set passed as argument. must be a model of concepts::CDigitalSet.
+     * @param set the set from which to build to DEC structure.
+     * @param add_border add border to the computed structure. For a precise definition see section \ref sectDECBorderDefinition.
+     */
+    template <typename TDigitalSet>
+    static
+    DiscreteExteriorCalculus<TDigitalSet::Point::dimension, TDigitalSet::Point::dimension, TLinearAlgebraBackend, TInteger>
+    createFromDigitalSet(const TDigitalSet& set, const bool add_border = true);
+
+    /**
+     * Create DEC structure from a range of signed n-cells, where n is the embedded dimension.
+     * Signed n-cells live in a ambient Khalimsky space whose dimension will determine the ambient dimension of the returned DEC structure.
```

Comment:
I would simply replace this sentence by something like:

"Signed n-cells may live in an ambient Khamlisky space with dimension greater than n."